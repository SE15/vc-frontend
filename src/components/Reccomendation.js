import {
    Button, Modal, ModalOverlay, ModalContent
    , ModalCloseButton, Text, useDisclosure, ModalFooter, Textarea
    , Avatar, Stack, Heading, Box
} from '@chakra-ui/react';
const Reccomendation = () => {
    const { isOpen, onOpen, onClose } = useDisclosure()

    return (
        <>
            <Button onClick={onOpen} bg="blueGreen.200">Post Reccomendation</Button>
            <Modal isOpen={isOpen} onClose={onClose} color="black.700" closeOnOverlayClick="false">
                <ModalOverlay />
                <ModalContent>
                    <Box bg="white.200" w="100%" p={4} color="white">
                        <stack>
                            <Heading color="black.400" fontSize="lg">
                                Post Reccomendation
                </Heading>
                            <br />
                            <Stack direction="row">
                                <Avatar name="Oshigaki Kisame" src="https://bit.ly/broken-link" />

                                <Text color="blue.100">
                                    Oshisidj Kisksls
            </Text>

                                <Textarea placeholder="Enter reccomendation" color="black.400" />
                                <ModalCloseButton />
                            </Stack>
                        </stack>
                        <ModalFooter>
                            <Button colorScheme="blue" mr={3} >
                                Post
            </Button>
                        </ModalFooter>
                    </Box>
                </ModalContent>

            </Modal>

        </>
    )
}
export default Reccomendation;