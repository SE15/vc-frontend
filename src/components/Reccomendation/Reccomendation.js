import { React } from "react"

import {Flex,Image, Text,Avatar, Stack, Heading, Box, Divider
} from '@chakra-ui/react';

const Reccomendation = props => {
    return (
        <>
        <Flex>
            <Box bg="white.200" w="50%" p={4} color="white">
                <Stack>  
                    <Stack direction="row">
                        if({props.postedImage}!=null){
                            <Image src={props.postedImage} size="100%" rounded="1rem" shadow="2xl" />
                        }
                        else{
                            <Avatar name={props.postedBy} src="https://bit.ly/broken-link" />
                        }
                        <Heading color="black.400" fontSize="lg">
                            {props.postedBy}
                        </Heading>
        
                    </Stack>
                    <Divider/>
                    <Text color="black.900" fontSize="md" textAlign="left">
                        {props.reccomendation}
                    </Text>
                </Stack>
            </Box>
            </Flex>
        </>
    )
}
export default Reccomendation