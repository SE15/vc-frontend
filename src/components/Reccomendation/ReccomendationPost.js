import  React, {useState} from "react"

import PropTypes from "prop-types"
import {
    Button, Modal, ModalOverlay, ModalContent, Image
    , ModalCloseButton, Text, useDisclosure, ModalFooter,
     Textarea, Avatar, Stack, Heading, Box
} from '@chakra-ui/react';
import { kPrimaryBlackLight,kSecondaryBlueLight } from '../../constants';
import { Component } from "react";
import Reccomendation from "./Reccomendation";

class ReccomendationPost extends Component{
    constructor(props){
        super(props);    
    }
    render(){
    return (
        <>
            <ReccomendationButton id="rec_button"visit={props.visit} name={props.name} />
        </>
    );
    }
}



class ReccomendationButton extends Component{
    constructor(props){
        super(props);
        this.state = {
            isLoggedIn: true,
            isOpen: false,
            isLoading: false,
            isreccomended: false,
            error: '',
            
        };
    }
    onClose=()=>{
        this.setState({
            ...this.state,
            isOpen: false
        }); 
    }
    onOpen= ()=>{
        this.setState({
            ...this.state,
            isOpen: true
        }); 
    }


    handleReccomendation= ()=>{
        this.setState({
            ...this.state,
            isLoading: true
        });
        try{
            //wait for update validation in db

            this.setState({
                ...this.state,
                isreccomended: true,
                isLoading: false
            });
        

        }catch{
            this.setState({
                ...this.state,
                isreccomended: false,
                isLoading: false,
                error: "Couldn't Post Reccomendation"
            });

        }
    }
    render(){
    if (this.props.visit == true && this.state.isLoggedIn == true && this.state.isreccomended==false) {

        return (
            <>
                <Box>
                    <Button id="post" onClick={()=>this.onOpen()} bg={kSecondaryBlueLight} isDisabled={false}>Post Reccomendation</Button>
                </Box>
                <Modal isOpen={this.state.isOpen} onClose={()=>onClose()} color="black.700" closeOnOverlayClick="false">
                    <ModalOverlay />
                    <ModalContent>
                        <Box bg={kPrimaryBlackLight} w="100%" p={4} color="white">
                            <Stack>
                                <Heading color="black.400" fontSize="lg">
                                    Post Reccomendation
                                </Heading>
                                <br />
                                <Stack direction="row">
                                    {this.props.cimage!=null?(
                                        <Image src={this.props.image}  borderRadius="full" boxSize="3rem" />
                                    ) :(
                                        <Avatar name={this.props.name} src="https://bit.ly/broken-link" />
                                    )
                                    }
                                    <Text color={kSecondaryBlueLight}>
                                        {this.props.name}
                                    </Text>
                   
                                    <Textarea placeholder="Enter reccomendation" color="black.400" />
                                    
                                    <ModalCloseButton />
                                </Stack>
                            </Stack>
                            <ModalFooter>
                                <Button colorScheme="blue" mr={3} onClick={()=>handleReccomendation()}>
                                    Post
                                </Button>
                            </ModalFooter>
                        </Box>
                    </ModalContent>
                </Modal>
            </>
        );
    } else {
        return (
            <Button onClick={()=>onOpen()} bg={kSecondaryBlueLight}isDisabled={true}>Post Reccomendation</Button>
        );
    }
    }   
}

export {
    ReccomendationPost,
    ReccomendationButton
} 