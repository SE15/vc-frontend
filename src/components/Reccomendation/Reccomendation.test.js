import React from 'react';

import {configure,mount,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

import Reccomendation from './Reccomendation'

configure({adapter:new Adapter()});

describe('<Reccomendation />',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper =shallow(<Reccomendation postedBy="Anne De Mel" />);
    });

    it('should display name and reccomendation according to layout',()=>{
        wrapper.setProps({reccomendation:"A recommendation letter should always be sincere and honest. Do not exaggerate the person's qualifications"})
        expect(wrapper.getElements()).toMatchSnapshot();
    })
});    