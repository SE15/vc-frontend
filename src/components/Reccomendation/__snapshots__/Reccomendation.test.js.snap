// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reccomendation /> should display name and reccomendation according to layout 1`] = `
Array [
  <React.Fragment>
    <Flex>
      <Box
        bg="white.200"
        color="white"
        p={4}
        w="50%"
      >
        <Stack>
          <Stack
            direction="row"
          >
            if(
            !=null)
            <Image
              rounded="1rem"
              shadow="2xl"
              size="100%"
            />
            else
            <Avatar
              name="Anne De Mel"
              src="https://bit.ly/broken-link"
            />
            <Heading
              color="black.400"
              fontSize="lg"
            >
              Anne De Mel
            </Heading>
          </Stack>
          <Divider />
          <Text
            color="black.900"
            fontSize="md"
            textAlign="left"
          >
            A recommendation letter should always be sincere and honest. Do not exaggerate the person's qualifications
          </Text>
        </Stack>
      </Box>
    </Flex>
  </React.Fragment>,
]
`;
